src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Client/Services/IdentityAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,56 @@
+using BlazorWasm_CSharp.Shared.Models;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace BlazorWasm_CSharp.Client
+{
+    public class IdentityAuthenticationStateProvider : AuthenticationStateProvider
+    {
+        private readonly HttpClient _client;
+        private ClaimsPrincipal _currentUser;
+
+        public IdentityAuthenticationStateProvider(HttpClient client)
+        {
+            _client = client;
+        }
+
+        public override async Task<AuthenticationState> GetAuthenticationStateAsync()
+        {
+            if (_currentUser == null)
+            {
+                await FetchUser();
+            }
+
+            return new AuthenticationState(_currentUser);
+        }
+
+        private async Task FetchUser()
+        {
+            var user = await _client.GetJsonAsync<UserInfo>("/User");
```

Comment:
The URL specified here needs to be `User`, not `/User`

Reply:
Yes, I forgot